File: query-dsl-java/src/main/java/com/example/querydsljava/web/PhoneController.java
package com.example.querydsljava.web;

import com.example.querydsljava.dao.entity.PhoneEntity;
import com.example.querydsljava.service.PhoneService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class PhoneController {

    private final PhoneService phoneService;

    @GetMapping("phone/{criteria}")
    public ResponseEntity<PhoneEntity> getPhone(@PathVariable String criteria) {
        return ResponseEntity.of(phoneService.getPhone(criteria));
    }
}

File: query-dsl-kotlin/src/main/kotlin/com/example/querydslkotlin/web/PhoneController.kt
package com.example.querydslkotlin.web

import com.example.querydslkotlin.service.PhoneService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody

@RestController
class PhoneController @Autowired constructor(
    private val phoneService: PhoneService
){

    @GetMapping("/phone/{criteria}")
    fun getPhone(@PathVariable("criteria") criteria: String) : ResponseEntity<StreamingResponseBody> {

        return ResponseEntity.ok()
            .contentType(MediaType.APPLICATION_OCTET_STREAM)
            .header("Content-disposition", "attachment; filename=redmi.csv")
            .body(phoneService.getPhone(criteria))
    }
}